nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/mail/Composer.java

```diff
@@ -88,32 +95,50 @@ public File getResourceFile(String key) {
 
     protected Mailer createMailer() {
         // first try the local configuration
-        org.nuxeo.common.Environment env = org.nuxeo.common.Environment.getDefault();
-        if (env != null) {
-            File file = new File(env.getConfig(), "mail.properties");
-            if (file.isFile()) {
-                Properties p = new Properties();
-                try {
-                    FileInputStream in = new FileInputStream(file);
-                    try {
-                        p.load(in);
-                        mailer = new Mailer(p);
-                    } finally {
-                        in.close();
-                    }
-                } catch (IOException e) {
-                    log.error("Failed to load mail properties", e);
-                }
-            }
+        // it was used by FakeSmtpMailServerFeature / EmbeddedAutomationClientTest#testSendMail - no runtime usage found
+        var properties = getMailProperties();
+        if (!properties.isEmpty()) {
+            mailer = new Mailer(properties);
         }
         // second try using JNDI
         if (mailer == null) {
-            String name = Framework.getProperty("jndi.java.mail", "java:/Mail");
+            String name = Framework.getProperty(CONFIGURATION_JNDI_JAVA_MAIL, DEFAULT_MAIL_JNDI_NAME);
             mailer = new Mailer(name);
         }
         return mailer;
     }
 
+    @NotNull
+    protected static Properties getMailProperties() {
+        File mailFile = getMailPropertiesFile();
+        if ((mailFile != null && MAIL_PROPERTIES.isEmpty()) || Framework.isTestModeSet()) {
+            synchronized (MAIL_PROPERTIES) {
+                if ((mailFile != null && MAIL_PROPERTIES.isEmpty()) || Framework.isTestModeSet()) {
```

Comment:
perhaps i am missing something, but do we want the second control (after the `synchronized` keyword)

Reply:
Yes, because if thread 1 acquires the lock and let thread 2 wait for it, when thread 2 will be un-lock it will executes what is inside the synchronized bloc.

Second if is to avoid that.